src/main/java/dev/openfeature/sdk/EventProvider.java

```diff
@@ -50,11 +87,18 @@ void detach() {
 
     /**
      * Emit the specified {@link ProviderEvent}.
-     * 
+     *
      * @param event   The event type
      * @param details The details of the event
      */
     public void emit(ProviderEvent event, ProviderEventDetails details) {
+        if (ProviderEvent.PROVIDER_ERROR.equals(event)) {
+            providerState = ProviderState.ERROR;
+        } else if (ProviderEvent.PROVIDER_STALE.equals(event)) {
+            providerState = ProviderState.STALE;
+        } else if (ProviderEvent.PROVIDER_READY.equals(event)) {
+            providerState = ProviderState.READY;
+        }
```

Comment:
Can you add a specific test for this? It relates to specification requirement [5.3.5](https://openfeature.dev/specification/sections/events/#requirement-535), so you can use our `@Specification` annotation for tracking compliance.